WIP PR: All Windows tools rebuilt for Linux

BDadmehr0 opened the pull request:
related issue: #149

BDadmehr0 committed:
Update linux.py

BDadmehr0 committed:
Update os_services.py

BDadmehr0 committed:
Update config.py

BDadmehr0 committed:
Update config.py

BDadmehr0 committed:
temp_file fix for linux & windows

BDadmehr0 committed:
config.py - Fix filename for linux

BDadmehr0 committed:
Update requirements.txt

BDadmehr0 committed:
req

_tkinter.TclError: bitmap "/home/dadmehr/Documents/GitHub/fishyboteso/fishy/icon.ico" not defined

BDadmehr0 committed:
Update terms_gui.py

BDadmehr0 committed:
fix terms_gui.py icon

change icon.ico to icon.png and change icon path to /fishy/gui/icon.png

BDadmehr0 committed:
icon fix

BDadmehr0 committed:
create shortcut Fix

BDadmehr0 committed:
fix

BDadmehr0 commented:
Run and install the app and prerequisites with **root access**.
The program uses libraries that are possible in the **root** directory.

BDadmehr0 committed:
beep & root | tmp acss

adsau59 commented on fishy/gui/splash.py:
```diff
@@ -32,7 +32,7 @@ def show(self, win_loc, q):
         top.title("Loading...")
         top.resizable(False, False)
-        top.iconbitmap(helper.manifest_file('icon.ico'))
+        top.iconphoto(True, tk.PhotoImage(file='fishy/icon.png'))
```
getting this on windows:

_tkinter.TclError: couldn't open "fishy/icon.png": no such file or directory

BDadmehr0 replied:
Translation: The method of creating an icon in Linux is different, so I can solve the problem with a simple condition.

adsau59 commented on fishy/helper/hotkey/hotkey_process.py:
```diff
@@ -63,7 +64,7 @@ def _event_loop(self):
                 callback = self._hotkeys[key]
                 if callback:
-                    playsound(helper.manifest_file("beep.wav"), False)
+                    playsound(helper.manifest_file("https://raw.githubusercontent.com/fishyboteso/fishyboteso/main/fishy/beep.wav"), False)  # Added comma here
```
y fetch it from web, instead of using the locally stored file?

BDadmehr0 replied:
No, this is a test, I am working on it

adsau59 commented on fishy/icon.ico:
.ico file is required to create shortcut in windows, can't really delete it
![explorer_56wTcglgL0](https://github.com/fishyboteso/fishyboteso/assets/10076396/880c9e7c-6fcc-4315-8340-4fa03454def6)

adsau59 commented on fishy/helper/hotkey/hotkey_process.py:
```diff
@@ -39,12 +40,12 @@ def stop():
-        self.inq = Queue()
-        self.outq = Queue()
+        self.inq = queue.Queue()
+        self.outq = queue.Queue()
 
         self._hotkeys: Dict[Key, Optional[Callable]] = dict([(k, None) for k in Key])
 
-        self.process = Process(target=process.run, args=(self.inq, self.outq))
+        self.process = Thread(target=process.run, args=(self.inq, self.outq))
```
using multithreading for hotkey lagged fishy really bad, thats why we used multiprocessing, we can't afford to use multithreading here...

why make this change though? I dont see any reason why it shouldn't work for linux... if it doesn't, maybe investigating why it doesn't rather than, switching it to multithreading

adam-saudagar reviewed (changes requested):
asked few queries, and requestred few changes

adsau59 commented on fishy/osservices/linux.py:
```diff
@@ -1,29 +1,101 @@
+import subprocess
+import re
+import os
+from Xlib import *
 from typing import Tuple, Optional
+from xdg.DesktopEntry import DesktopEntry
```
dont these need to be in requirement.txt?

adsau59 commented on fishy/osservices/linux.py:
```diff
@@ -1,29 +1,101 @@
+        window_name = window.get_wm_name()
+        return window_name == "Elder Scrolls Online"
+
+    def get_monitor_rect():
```
missing self in parameter

adsau59 commented on fishy/osservices/linux.py:
```diff
@@ -1,29 +1,101 @@
+import subprocess
+import re
+import os
+from Xlib import *
```
dont import everything, import the things u need to use... that way we can see which pakage is providing which variable

so change this to `from Xlib import display` if you aren't using anything else from the package

adam-saudagar reviewed (changes requested):
changes requested

SemjonWilke commented on requirements.txt:
```diff
@@ -2,7 +2,7 @@ urllib3
 numpy
 opencv_python
-Pillow
+pillow
```
U sure? It really looks like it should be Pillow on linux too
![grafik](https://github.com/fishyboteso/fishyboteso/assets/1882648/d8558390-b9dc-4168-9eb1-d2f11e7b43da)
https://pillow.readthedocs.io/en/stable/installation.html

SemjonWilke commented on requirements.txt:
```diff
@@ -15,4 +15,4 @@ playsound==1.2.2
 pyautogui
-mss
\ No newline at end of file
+mss
```
ws

SemjonWilke commented on fishy/osservices/linux.py:
tl;dr
btw linux users will come in hordes to fix it themselves lol

SemjonWilke reviewed (changes requested):
Is not starting on windows, needs a rebase on main and some fixes are required.
Little more wiggling and wobbling needed until this can go in.

SemjonWilke commented on fishy/osservices/linux.py:
```diff
@@ -1,29 +1,101 @@
+            window_rect.y + window_geometry.height
+        )
 
-    def get_game_window_rect(self) -> Optional[Tuple[int, int, int, int]]:
```
not providing this function definition will likely create problems

SemjonWilke commented on fishy/osservices/linux.py:
```diff
@@ -1,29 +1,101 @@
+        if window_id is None:
+            return None
 
-    def get_monitor_rect(self):
```
not providing this function definition will likely create problems

SemjonWilke commented on fishy/osservices/linux.py:
```diff
@@ -1,29 +1,101 @@
 
     def get_documents_path(self) -> str:
-        pass
+        return os .path.join(os.path.expanduser('~'), "Documents")
```
os\ .

BDadmehr0 committed:
fix linux.py bugs

BDadmehr0 committed:
Update linux.py

BDadmehr0 committed:
Update linux.py

BDadmehr0 committed:
Create __init__.pyc

BDadmehr0 committed:
Create install.sh script installer

script installer for Linux OS manjaro, unbuntu, fedora

BDadmehr0 committed:
fix app icon

support windows and linux with check os name

BDadmehr0 committed:
add file to MAINFEST

BDadmehr0 committed:
fix icon & fixed hide_tirminal & add update command to install.sh

Changing the way to hide the terminal in Linux with the vm ctrl method

The required prerequisites are available in the install.sh installation file for easier installation on Linux

BDadmehr0 committed:
install.sh update

BDadmehr0 committed:
Merge branch 'main' into linux

BDadmehr0 committed:
Update linux.py

BDadmehr0 committed:
Update splash.py

BDadmehr0 committed:
Update hotkey_process.py

BDadmehr0 committed:
Update MANIFEST.in
